Samba generates a profile sniplet with permissions for all shares at 
start using the update-apparmor-samba-profile script.

This patch includes the autogenerated profile sniplet it in the smbd 
profile. It also creates a dummy profile sniplet to avoid "file not 
found" errors when AppArmor is started before samba was started.

References: https://bugzilla.novell.com/show_bug.cgi?id=688040


Signed-off-by: Christian Boltz <apparmor@cboltz.de>

=== added file 'profiles/apparmor.d/local/usr.sbin.smbd-shares'
--- profiles/apparmor.d/local/usr.sbin.smbd-shares	1970-01-01 00:00:00 +0000
+++ profiles/apparmor.d/local/usr.sbin.smbd-shares	2011-10-19 09:40:05 +0000
@@ -0,0 +1,2 @@
+# This file will be replaced by rules for all samba shares at samba start.
+# Do not edit!

=== modified file 'profiles/apparmor.d/usr.sbin.smbd'
--- profiles/apparmor.d/usr.sbin.smbd	2011-08-27 18:50:42 +0000
+++ profiles/apparmor.d/usr.sbin.smbd	2011-10-19 09:37:04 +0000
@@ -55,6 +55,10 @@
 
   @{HOMEDIRS}/** lrwk,
 
+  # permissions for all configured shares
+  # autogenerated by update-apparmor-samba-profile at samba start
+  #include <local/usr.sbin.smbd-shares>
+
   # Site-specific additions and overrides. See local/README for details.
   #include <local/usr.sbin.smbd>
 }

